### ISSUE TITLE
scrollElements not sync with angular partials

### ISSUE THREAD

**Issue by KBizien** [BrowserSync/browser-sync#1141]
Hi there,

### Issue details

I try to use browsersync as a testing tool for my angular application. I've got a problem with the option scrollElements who seems to not syncing elements who are in angular templates.
For exemple, I want to sync the scroll on a div with the class .test. Browsersync synchronise the scroll event if this div is in index.html (the layout), but not if this div is in a template call with ui-router.

### Steps to reproduce/test case

Launch the gulp task from a terminal and test the scroll sync between Chrome and Firefox.

### Please specify which version of Browsersync, node and npm you're running

- Browsersync [  2.13.0  ]
- Node        [  5.11.0  ]
- Npm         [  3.8.6  ]

### Affected platforms

- [ ] linux
- [ ] windows
- [x] OS X
- [ ] freebsd
- [ ] solaris
- [ ] other _(please specify which)_

### Browsersync use-case

- [ ] API
- [x] Gulp
- [ ] Grunt
- [ ] CLI

### for all other use-cases, (gulp, grunt etc), please show us exactly how you're using Browsersync

```
var gulp = require("gulp");
var browserSync = require('browser-sync');
var browserSyncSpa = require('browser-sync-spa');

gulp.task('serve', function() {

  browserSync.use(browserSyncSpa({
    selector: '[ng-app]',
    history: {
      index: './index.html'
    }
  }));

  browserSync.init({
    server: {
      baseDir: './',
    },
    scrollElements: ['.test'],
    open: false,
    reloadOnRestart: true
  });
});

```

My app architecture:

App
----node_modules
----partials
--------page1.html
--------page2.html
----app.js
----gulpfile.js
----index.html
----package.json

Thanks for your help

**Comment by shakyShane** [comment]
Thanks for the report, this is too difficult to debug without a sample project the issue - if you could provide that I may be able to assit :)

**Comment by KBizien** [comment]
Yes I understand, I pushed a sample project here [](https://github.com/KBizien/browsersync-angular)
You can clone this repo and run `npm i && npm start` - Browsersync config is in gulpfile.js, there is a .test div in index.html and partials/page2.html. If you open this sample app at localhost:3000 in two different browsers, you'll see the .test div in index.html is sync but not .test div in partials/page2.html.

Thanks a lot :)

**Comment by tuhoojabotti** [comment]
I'm also trying to use BrowserSync with Angular Material and I have a md-content element with the attribute "md-scroll-y" so I added `scrollElements: ['[md-scroll-y]']`, but it does not work. I'm guessing because BrowserSync tries to find the elements too soon before they are rendered by Angular.

**Comment by KBizien** [comment]
Hi there,
I call again this issue, any solution ?

Thanks for your help

**Comment by tuhoojabotti** [comment]
I ended up building my own plugin for BrowserSync. If you're interested I could post a gist of it.

**Comment by KBizien** [comment]
Oh yeah! It would be great :)

Thanks a lot

**Comment by tuhoojabotti** [comment]
@KBizien Here you go: https://gist.github.com/tuhoojabotti/b35768ba4daec99843e0c5cbab1885c5

To activate it you have to tell BrowserSync to use it in your build script:

```
browserSync.use({
	plugin() {},
	hooks: {
		// The 'client:js' hook will add your own JS file to the one used by
		// BrowserSync in the browser.
		'client:js': fs.readFileSync(`${__dirname}/browsersync-scroll-plugin.js`)
	}
});
```

Note: Currently this only restores scroll position when BrowserSync refreshes your page, but you should be able to extend it to sync scroll position to other instances pretty easily.
